libs/core/src/lib/utils/functions/browser-detection.ts

```diff
@@ -0,0 +1,15 @@
+export class BrowserDetection {
+    /** @hidden */
+    static isIE(): boolean {
+        const ua = window.navigator.userAgent; // Check the userAgent property of the window.navigator object
+        const msie = ua.indexOf('MSIE '); // IE 10 or older
+        const trident = ua.indexOf('Trident/'); // IE 11
+
+        return msie > 0 || trident > 0;
+    }
+
+    /** @hidden */
+    static isFirefox(): boolean {
+        return navigator.userAgent.indexOf('Firefox') !== -1
+    }
+}
```

Comment:
Since we started using `angular/cdk` for some internal staff maybe it would be better to use [angular/cdk/platform](https://material.angular.io/cdk/platform/api) for such purposes?

Reply:
@dimamarksman I removed this file and added `cdk/platform` browser detection. Thanks for pointing it